pilot/cmd/pilot-agent/status/server.go

```diff
@@ -23,33 +23,54 @@ import (
 	"os"
 	"sync"
 
+	"time"
+
 	"istio.io/istio/pilot/cmd/pilot-agent/status/ready"
 	"istio.io/istio/pkg/log"
 )
 
 const (
+	// readyPath is for the pilot agent readiness itself.
 	readyPath = "/healthz/ready"
+	// appReadyPath is the path for the application after injecting.
+	appReadinessPath = "/app/ready"
```

Comment:
Please add `server_test.go`  with a fake application and drive it like a prober.
This package is missing tests.

Reply:
Done.